### PR TITLE
Quarterdeck Project Info Endpoints

### PR DESCRIPTION
**Pull request by bbengfort**
### Scope of changes

Implements a project list and project detail endpoint so that Tenant can retrieve information about the number of active and revoked API keys. 

Fixes SC-16493

### Type of change

- [x] new feature
- [ ] bug fix
- [ ] documentation
- [ ] testing
- [ ] technical debt
- [ ] other (describe)

### Acceptance criteria

Does this give you the information you need for the statistics endpoint? Do you need to differentiate between active (used within the last 30 days) and not-revoked api keys?

### Author checklist

- [x] I have manually tested the change and/or added automation in the form of unit tests or integration tests
- [ ]  I have updated the dependencies list
- [ ]  I have recompiled and included new protocol buffers to reflect changes I made
- [x]  I have added new test fixtures as needed to support added tests
- [x]  Check this box if a reviewer can merge this pull request after approval (leave it unchecked if you want to do it yourself)
- [ ]  I have moved the associated Shortcut story to "Ready for Review"

### Reviewer(s) checklist

- [ ] Any new user-facing content that has been added for this PR has been QA'ed to ensure correct grammar, spelling, and understandability.
- [ ] Are there any TODOs in this PR that should be turned into stories?

### CODE REVIEW

File: pkg/quarterdeck/db/models/orgs.go
Comment (bbengfort) [inline pkg/quarterdeck/db/models/orgs.go:1]: A bit of reorganization to make projects easier to find in the codebase.

File: pkg/quarterdeck/orgs.go
Comment (bbengfort) [inline pkg/quarterdeck/orgs.go:1]: A bit of reorganization to make projects easier to find in the codebase.

File: pkg/quarterdeck/db/models/orgs_test.go [hunk 1]
```diff
@@ -100,15 +99,15 @@ func (m *modelTestSuite) TestListOrgs() {
 	require.NotNil(org.ID)
 	require.NotNil(org.Name)
 	require.NotNil(org.Domain)
-	require.Equal(2, org.ProjectCount(), "expected 2 projects for organization Testing")
+	require.Equal(3, org.ProjectCount(), "expected 2 projects for organization Testing")
```
Comment (pdeziel) [inline pkg/quarterdeck/db/models/orgs_test.go:102]:
```suggestion
	require.Equal(3, org.ProjectCount(), "expected 3 projects for organization Testing")
```

File: pkg/quarterdeck/db/models/projects.go [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+	SELECT op.organization_id, op.project_id, op.created, op.modified, count(k.id) as apikeys_count
+		FROM organization_projects op
+		LEFT JOIN api_keys k ON op.organization_id=k.organization_id AND op.project_id = k.project_id
+	GROUP BY op.organization_id, op.project_id),
+project_counts AS (
+	SELECT p.*, count(r.id) as revoked_count
+		FROM projects p
+		LEFT JOIN revoked_api_keys r ON p.organization_id=r.organization_id AND p.project_id=r.project_id
+	GROUP BY p.organization_id, p.project_id)
+	SELECT * FROM project_counts`
```
Comment (pdeziel) [inline pkg/quarterdeck/db/models/projects.go:127]: The line breaks and indents definitely help with readability 👍

File: pkg/quarterdeck/db/models/projects.go [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+	var tx *sql.Tx
+	if tx, err = db.BeginTx(ctx, &sql.TxOptions{ReadOnly: true}); err != nil {
+		return false, err
+	}
+	defer tx.Rollback()
+
+	if ok, err = op.exists(tx); err != nil {
+		return ok, err
+	}
+	tx.Commit()
```
Comment (pdeziel) [inline pkg/quarterdeck/db/models/projects.go:101]: `tx.Commit()` returns an error right?

File: pkg/quarterdeck/orgs.go
Comment (pdeziel) [inline pkg/quarterdeck/orgs.go:1]: Nice idea to move the project methods to their own file.

File: pkg/quarterdeck/projects.go [hunk 4]
```diff
@@ -0,0 +1,271 @@
...
+			sentry.Warn(c).Err(err).Str("next_page_token", query.NextPageToken).Msg("could not parse next page token")
+			c.JSON(http.StatusBadRequest, api.ErrorResponse(err))
+			return
+		}
+	} else {
+		prevPage = pagination.New("", "", int32(query.PageSize))
+	}
+
+	if projects, nextPage, err = models.ListProjects(c.Request.Context(), orgID, prevPage); err != nil {
+		sentry.Error(c).Err(err).Msg("could not parse orgID from claims")
```
Comment (pdeziel) [inline pkg/quarterdeck/projects.go:66]: Should this have a different error message?

File: pkg/quarterdeck/projects.go [hunk 5]
```diff
@@ -0,0 +1,271 @@
...
+		Projects: make([]*api.Project, 0, len(projects)),
+	}
+
+	for _, project := range projects {
+		out.Projects = append(out.Projects, project.ToAPI())
+	}
+
+	if nextPage != nil {
+		if out.NextPageToken, err = nextPage.NextPageToken(); err != nil {
+			sentry.Error(c).Err(err).Msg("could not parse orgID from claims")
```
Comment (pdeziel) [inline pkg/quarterdeck/projects.go:82]: Should this have a different error message?

**Review by pdeziel — COMMENTED** [review]
Thanks for adding the endpoints, should definitely make things easier on the Tenant side. I didn't mark approve yet because I had a few comments, but they should be pretty straightforward.

File: pkg/quarterdeck/db/models/projects.go (hunk 2 above)
Comment (bbengfort) [reply pkg/quarterdeck/db/models/projects.go:127]: It's a pretty hefty query; that's for sure.

File: pkg/quarterdeck/db/models/projects.go (hunk 3 above)
Comment (bbengfort) [reply pkg/quarterdeck/db/models/projects.go:101]: It does, but it doesn't matter that much in a readonly transaction.

File: pkg/quarterdeck/projects.go (hunk 4 above)
Comment (bbengfort) [reply pkg/quarterdeck/projects.go:66]: good catch!

**Comment by bbengfort** [comment]
@pdeziel good catch with those copy and paste errors - I've made the fixes.

**Review by pdeziel — APPROVED** [review]
Looks good, thanks for making the changes!

**Pull request merged**